src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,131 @@
+package duke.task;
+
+import duke.exception.DukeException;
+import duke.ui.UI;
+
+import java.util.ArrayList;
+
+/**
+ * Responsible for managing all the tasks in the list. This class will use an ArrayList inside it.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    public int size() {
+        return tasks.size();
+    }
+
+    void add(Task task) {
+        tasks.add(task);
+    }
+
+    public Task get(int index) {
+        return tasks.get(index);
+    }
+
+    public void changeDone(String line, UI ui, TaskList tasks) {
```

Comment:
changeDone sounds a little bit weird, what about setDone?

Reply:
Yes, you're right! Thanks for the suggestion!